### DISCUSSION TITLE
How to distribute .NET 6 runtime via MSIX?

### DISCUSSION THREAD

**Discussion by MarkIngramUK** [dotnet/runtime#69274] [Q&A] [answered]
We have a WPF .NET Framework 4.8 application which we're currently distributing via MSIX. I'm looking at migrating to .NET 6, but I can't find a definitive answer on how to ship the .NET 6 runtime via MSIX. I don't want to trigger an install the first time the app runs, so I want the runtime installed at the same time as our app. How can I achieve that? We manually create the MSIX via [Packaging API](https://docs.microsoft.com/en-us/windows/win32/appxpkg/interfaces).

**Comment by bjornen77** [comment]
This is not an direct answer to you question, but maybe the self contained publish feature in .NET 6 could solve your problem? By using that feature you do not need to install the .NET 6 runtime on the host. The application will contain everything needed including the runtime itself. You could also publish everything to a single .exe file if wanted.

**Reply by vitek-karas** [reply]
@bjornen77 is correct. Currently the only way to do this is to publish the app as self-contained.

**Reply by MarkIngramUK** [reply]
Thanks @bjornen77 and @vitek-karas . How do I manually publish a self-contained app? Currently we have an output directory full of .exe and .dll files, which we then add to the MSIX package (using packaging APIs). Where do I get the .NET 6 "self-contained" runtime, to use with our manual MSIX packaging approach?

**Reply by vitek-karas** [reply]
`dotnet publish --self-contained -r win-x64` - more details here: https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained

**Comment by danmoseley** [comment]
Cc @eerhardt

**Reply by TonyValenti** [reply]
I am definitely a fan of single file publishing. I've built a library that uses that capability to enable silent in-place updates of its own exe.

**Reply by eerhardt** [reply]
Just an FYI, you can do single-file publishing, but still be “framework dependent”. So the shared runtime in ProgramFiles will be used with the single file executable.

**Comment by teo-tsirpanis** [comment, marked as answer]
Duplicate of #100880.
